src/context/auth/authContext.js

```diff
@@ -0,0 +1,68 @@
+import { createContext, useContext, useReducer } from "react";
+import { loginApi, signupApi } from "../../services/auth_services";
+import { reducer } from "./authReducer";
+
+const AuthContext = createContext();
+const AuthContextProvider = ({ children }) => {
+  const user = JSON.parse(localStorage.getItem("user"));
+  const intialState = user ?? {
```

Comment:
I am not able to understand use of ?? here

Reply:
Read [this](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator)